### PR TITLE
fix: support transitive npm aliased deps in pnpm lockfile

### PR DESCRIPTION
**Pull request by gregmagolan**
Fixes https://github.com/aspect-build/rules_js/issues/164

```
.../rules_js/bazel-bin/node_modules/.aspect_rules_js/plotly.js@2.12.1/node_modules/regl -> /private/var/tmp/_bazel_greg/7aed427a991d86f99332ab79b9b11780/execroot/aspect_rules_js/bazel-out/darwin-fastbuild/bin/node_modules/.aspect_rules_js/@plotly+regl@2.1.2/node_modules/@plotly/regl
```

### CODE REVIEW

File: .bazelignore [hunk 1]
```diff
@@ -1,2 +1,7 @@
 e2e/
-examples/node_modules/
+examples/assert_lockfile_up_to_date/node_modules/
```
Comment (alexeagle) [inline .bazelignore:2]: nit: could add a test that this file and pnpm-workspace.yaml agree on the node_modules folder locations

**Review by alexeagle — APPROVED** [review]

File: .bazelignore (hunk 1 above)
Comment (gregmagolan) [reply .bazelignore:2]: agreed; it is annoying that it is can easily fall out of date; users should do the same

Comment (gregmagolan) [reply .bazelignore:2]: will make an issue to do as a follow-up

Comment (gregmagolan) [reply .bazelignore:2]: https://github.com/aspect-build/rules_js/issues/196

**Pull request merged**